cmake/pcre.cmake

```diff
@@ -85,6 +85,7 @@ MACRO (CHECK_PCRE)
       PKG_CHECK_MODULES(PCRE libpcre2-8)
       # in case pkg-config or libpcre2-8.pc is not installed:
       CHECK_LIBRARY_EXISTS(pcre2-8 pcre2_match_8 "${PCRE_LIBRARY_DIRS}" HAVE_PCRE2_MATCH_8)
+      link_directories(${PCRE_LIBRARY_DIRS})
```

Comment:
does it work "in case pkg-config or libpcre2-8.pc is not installed" ?

Reply:
I am not sure it's a big problem — not working "in case pkg-config or libpcre2-8.pc is not installed".

But the comment two lines above explicitly says `CHECK_LIBRARY_EXISTS` is to make the code work in this case. If it doesn't — the comment should be removed or changed and `CHECK_LIBRARY_EXISTS` might need to be removed if it's redundant (but perhaps it's still needed for `build_depends.cmake` — then it should stay and the comment should say that it's for build dependencies)

Reply:
@vuvova Sorry for the hiatus. I feel we should still keep `CHECK_LIBRARY_EXISTS()` and make it compatible when pc files are not available in edge cases. I've reworked the PR and use `FIND_LIBRARY()` when it's missing and it should work for both case now.